lib_xud/src/core/included/XUD_RxData.S

```diff
@@ -24,110 +24,131 @@
 // r11: Expected CRC
 
 Pid_Bad_RxData:
-    ldaw      r10, dp[PidJumpTable]
-    {ldw      r11, r10[r4];         ldc r8, 16}
-    {bau      r11;                  setpsc      res[RXD], r8}   // XUD_CrcAddrCheck.S requires 16 in r8
+    {ldc      r8, 16;               ldw r10, sp[STACK_PIDJUMPTABLE]} // XUD_CrcAddrCheck.S requires 16 in r8
+xta_ep_Token_EPCRC_doRXData:
+    {setpsc   res[RXD], r8;         ldw r11, r10[r4]}
+xta_targetA_Pid_Reserved:
+xta_targetA_Pid_Out:
+xta_targetA_Pid_Ack:
+xta_targetA_Pid_Data0:
+xta_targetA_Pid_Ping:
+xta_targetA_Pid_Sof:
+xta_targetA_Pid_Sof_NoChan:
+xta_targetA_Pid_Nyet:
+xta_targetA_Pid_Data2:
+xta_targetA_Pid_Split:
+xta_targetA_Pid_In:
+xta_targetA_Pid_Nak:
+xta_targetA_Pid_Data1:
+xta_targetA_Pid_Pre:
+xta_targetA_Pid_Setup:
+xta_targetA_Pid_Stall:
+xta_targetA_Pid_Datam:
+xta_targetA_Pid_Bad:
+    {bau      r11;                  clre}
 
 doRXData:
+xta_ep_Token_PID_doRXData:
     inpw        r4, res[r0], 8                                  // Input PID
+    {eeu        res[r0];            ldw r8, sp[STACK_RXA_PORT]}
 
-    ldw         r8, sp[STACK_RXA_PORT]
 #ifdef __XS2A__
     // If pid != DATAx then jump and handle as a token. DATA0, DATA1, DATA2 & MDATA all of the form 0bxx11.
     // This is a fast alternative to a "timeout"
     // Note, this doesn't check that PID[0:3] = ~PID[4:7] - which is an issue for XS3
     {mkmsk r11, 2;                  shr         r4, r4, 24}
     and         r11, r11, r4                                    // Store PID into EP structure,
     eq          r11, r11, 3
-    bf          r11, Pid_Bad_RxData
+    {eeu        res[r8];            bf          r11, Pid_Bad_RxData}
 #else
-    {shr        r4, r4, 24;         ldw     r11, sp[STACK_PIDJUMPTABLE_RXDATA]}
+    {shr        r4, r4, 24;         ldw r11, sp[STACK_PIDJUMPTABLE_RXDATA]}
     ldw         r11, r11[r4]
-    bau         r11
+xta_targetB_Pid_Data0_RxData:
+xta_targetB_Pid_Data2_RxData:
+xta_targetB_Pid_Data1_RxData:
+xta_targetB_Pid_Datam_RxData:
+xta_targetB_Pid_Bad_RxData:
+    {eeu        res[r8];            bau         r11}
 #endif
 
 Pid_Datam_RxData:
 Pid_Data0_RxData:
 Pid_Data1_RxData:
 Pid_Data2_RxData:
-    {stw         r4, r3[6];         setsr 1}                    // Store PID into EP structure
-
+    {stw        r4, r3[6];          sub r4, r1, 4}          // Store PID into EP structure
+                                                            // Init pointer to index -1
 GotRxPid:
-    {eeu        res[r8];            mkmsk r4, 32}               // Enable events on RxA
-                                                                // Init buffer index to -1
+xta_targetA_RxALow:
+xta_targetA_NextRxWord:
+    waiteu              // Enable events on RxA and RxD, only needed to handle short words
```

Comment:
Technically // ~Enable~ Wait for events..

Reply:
It does set the eeble of the thread but yeah, not the resources